### PR TITLE
Don't call atexit on closed sessions

### LINKED ISSUE #2: Shouldn't call atexit on closed sessions.
If there isn't a `self._session` we shouldn't try to call it's methods.

### PR DESCRIPTION
**Pull request by frankwiles**
Fixes #2

### CODE REVIEW

**Comment by fcurella** [comment]
Will need to be backported to the  [`releases/v1.1.x`](https://github.com/genericclient/genericclient-aiohttp/tree/releases/v1.1.x) branch

File: genericclient_aiohttp/__init__.py [hunk 1]
```diff
@@ -208,7 +208,8 @@ def get_or_create_session(self):
         return self
 
     async def __aexit__(self, *args, **kwargs):
-        await self._session.__aexit__(*args, **kwargs)
+        if self._session is not None and not self._session.closed:
```
Comment (fcurella) [inline genericclient_aiohttp/__init__.py:212]: shuold this be an `or` instead of `and`?

Comment (fcurella) [reply genericclient_aiohttp/__init__.py:212]: nevermind, we could have `_session` set but already closed.

File: genericclient_aiohttp/__init__.py [hunk 2]
```diff
@@ -208,7 +208,9 @@ def get_or_create_session(self):
         return self
 
     async def __aexit__(self, *args, **kwargs):
-        await self._session.__aexit__(*args, **kwargs)
+        self.get_or_create_session()
```
Comment (fcurella) [inline genericclient_aiohttp/__init__.py]: this line need to be moved to the `__aenter__` method

**Review by fcurella — CHANGES_REQUESTED** [review]

File: genericclient_aiohttp/__init__.py (hunk 2 above)
Comment (frankwiles) [reply genericclient_aiohttp/__init__.py]: Doh yeah typo

**Review by fcurella — APPROVED** [review]

**Pull request merged**